middle_end/flambda/simplify/simplify_set_of_closures.ml

```diff
@@ -517,12 +513,18 @@ let simplify_function context ~used_closure_vars ~shareable_constants
         ~code_id:new_code_id
         ~is_tupled:(FD.is_tupled function_decl)
     else
-      (* We need to manually specify the cost metrics to use to ensure that
-         they are the one of the body after simplification. *)
+      let code =
+        match Rebuilt_static_const.to_const code with
+        | Some const ->
+          begin match Static_const.to_code const with
+          | Some code -> code
+          | None -> assert false
+          end
+        | None -> assert false
+      in
```

Comment:
This is a bit ugly.  How about changing the conditional on line 509 to just call `Rebuilt_static_const.to_const`.  Then we can match on that; if it's `None` we call `T.create_non_inlinable_function_declaration` as currently.  If it's `Some` we will have to call `Static_const.to_code`, but this shouldn't yield `assert false` if it doesn't work -- use `Misc.fatal_errorf` instead.